azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/AvailabilitySets.java

```diff
@@ -3,25 +3,19 @@
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsDeletingByGroup;
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsGettingByGroup;
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsListingByGroup;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsCreating;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsDeleting;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsListing;
+import com.microsoft.azure.management.resources.fluentcore.collection.ResourcesInGroup;
 
 /**
  * Entry point to availability set management API.
  */
-public interface AvailabilitySets extends
-        SupportsListing<AvailabilitySet>,
+public interface AvailabilitySets extends 
+        ResourcesInGroup<AvailabilitySet, AvailabilitySet.DefinitionBlank>,
         SupportsListingByGroup<AvailabilitySet>,
         SupportsGettingByGroup<AvailabilitySet>,
-        SupportsCreating<AvailabilitySet.DefinitionBlank>,
-        SupportsDeleting,
         SupportsDeletingByGroup {
-    /**
-     * Entry point to availability set management API within a specific resource group.
-     */
-    interface InGroup extends
-            SupportsListing<AvailabilitySet>,
-            SupportsCreating<AvailabilitySet.DefinitionCreatable>,
-            SupportsDeleting {}
+    
+    
+    public interface InGroup 
+        extends ResourcesInGroup<AvailabilitySet, AvailabilitySet.DefinitionAfterGroup> {
```

Comment:
that;s what it said originally, but the reason I created this extra layer of indirection is because otherwise, your InGroup impl needs to know what follows the withGroup for a given model. And that varies. So if you ned to change the order of required params or add more required params after, you have to change both the interface and the impl. 
By establishing this convention that all interfaces have this DefinitionAfterGroup after group, we can have the ingroup impls only refer to that consistently. That way when your required parameter sequence changes, you only change it in the interface, not the impl.